### PR TITLE
fix: remove duplicated validation

### PR DESCRIPTION
**Pull request by hezhizhen**
**1. Issue, if available:**

**2. Description of changes:**
It removes duplicated validation for options.

**3. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmd/controller/main.go [hunk 1]
```diff
@@ -60,10 +60,6 @@ func init() {
 }
 
 func main() {
```
Comment (ellistarn) [inline cmd/controller/main.go:62]: Can you add a reference to where the other source of validation is?

Comment (hezhizhen) [reply cmd/controller/main.go:62]: Options are validated when they are parsed. https://github.com/aws/karpenter/blob/75afd6006eeb60669995419d82a412e1e174ccd4/pkg/utils/options/options.go#L37-L39

Comment (hezhizhen) [reply cmd/controller/main.go:62]: @ellistarn Shall we resolve it, or add some comments to the code?

**Comment by jsoref** [comment]
Fwiw, check-spelling doesn't currently respond to `/retest`, a force-push (or rebase + force-push) would result in a pass.

**Review by ellistarn — APPROVED** [review]
Thank you!

**Pull request merged**